squarelet/users/views.py

```diff
@@ -148,22 +158,65 @@ class UserListView(LoginRequiredMixin, ListView):
 
 
 class LoginView(AllAuthLoginView):
-    """Subclass of All Auth Login View to add redirect ability for failed auth tokens
-
-    If the url_auth_token parameter is still present, it means the auth token failed
-    to authenticate the user.  Redirect them to the nested next parameter instead of
-    asking them to login
-    """
+    def dispatch(self, request, *args, **kwargs):
+        """Override form_valid to provide onboarding steps, if needed"""
+        # handle the request as usual, but hold onto the response
+        response = super().dispatch(request, *args, **kwargs)
+        # when logging in, a POST is sent
+        is_post = request.method == "POST"
+        # if the login succeeded, then the user is authenticated
+        user = self.request.user
+        if is_post and user.is_authenticated:
+            next_url = self.get_success_url()
+            intent = request.GET.get("intent")
+            # Check the user's account status
+            if not has_verified_email(user):
+                # If the user has not verified their email,
+                # send a new confirmation email and redirect
+                send_email_confirmation(request, user, False, user.email)
+                url = (
+                    reverse("login_confirm_email") + f"?next={next_url}"
+                    if next_url
+                    else reverse("login_confirm_email")
+                )
+                if intent:
+                    url += f"&intent={intent}"
```

Comment:
This will break if there is no `next_url` since the & will need to be a ? - can possibly use `urllib.parse.urlencode` to do this cleanly